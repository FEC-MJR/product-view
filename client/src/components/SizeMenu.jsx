import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
`;

const StyledMenu = styled.select`
 width: 100%;
 padding: 10px 30px 10px 20px;
 border: 2px solid rgba(0, 0, 0, .15);
 background: #fff;
 border-radius: 2px;
`;

class SizeMenu extends React.Component {

  constructor(props) {
    super(props)

    this.state = {
      isLoaded: false,
      sizes: this.props.data.size,
      hasSizes: false
    }
  }

  componentDidMount() {
    if (this.state.sizes[0] !== 'One Size') {
      this.setState({
        isLoaded: true,
        hasSizes: true
      })
    } else {
      this.setState({
        isLoaded: true
      })
    }
  }

  render() {

    const ItemSize = this.state.sizes.map((item, index) => (
      <option key={index}>{item}</option>
    ))

    if (this.state.hasSizes === true) {
      return(
        <Wrapper>
          <StyledMenu>
            <option defaultValue="Select a size" hidden>Select a size</option>
            {ItemSize}
          </StyledMenu>
        </Wrapper>
      )
    } else {
      return(
        <Wrapper>
          <StyledMenu>
            <option defaultValue="One Size" hidden>One Size</option>
            {ItemSize}
          </StyledMenu>
        </Wrapper>
      )
    }
  }
}

export default SizeMenu;