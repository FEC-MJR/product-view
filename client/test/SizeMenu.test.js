import React from 'react'
import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer'
import SizeMenu from  '../src/components/SizeMenu.jsx'

describe("Component testing for <SizeMenu/>", () => {

  const props = {
    data:  {
      name: "KANKEN LAPTOP 13\"",
      type: "backpack",
      description: "Sed eu eros. Nam consequat dolor vitae dolor. Donec fringilla. Donec feugiat metus sit",
      userRating: 5,
      totalRatings: 6,
      price: 110.00,
      gender: "Unisex",
      style: "Style # F27171",
      size: ["One Size"],
      color: "166 - Acorn",
      image: ["https://hrphxfec.s3-us-west-1.amazonaws.com/Kanken+Laptop+13/kanken_acorn.jpg"]

    }
  }

  test("<SizeMenu/> renders", () => {

    const wrapper = shallow(<SizeMenu {...props}/>)
    expect(wrapper)
  })

  test("Placeholder value should equal 'One Size'", () => {

    const wrapper = shallow(<SizeMenu {...props}/>)
    expect(wrapper.find('.placeholder').text()).toBe('One Size')
  })

  test("<SizeMenu/> snapshot", () => {

    const tree = renderer.create(<SizeMenu {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})