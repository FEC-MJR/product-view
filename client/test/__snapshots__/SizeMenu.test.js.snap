// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing for <SizeMenu/> <SizeMenu/> snapshot 1`] = `
<div>
  <select
    className="sc-AxjAm fGLTcM"
  >
    <option
      className="placeholder"
      defaultValue="One Size"
      hidden={true}
    >
      One Size
    </option>
    <option>
      One Size
    </option>
  </select>
</div>
`;
